src/domain/wallets/payment-input-validator.ts

```diff
@@ -0,0 +1,50 @@
+import { checkedToSats } from "@domain/bitcoin"
+import {
+  SelfPaymentError,
+  InvalidWalletId,
+  InvalidAccountStatusError,
+} from "@domain/errors"
+import { AccountStatus } from "@domain/accounts"
+
+export const PaymentInputValidator = (
+  getWalletFn: PaymentInputValidatorConfig,
```

Comment:
why passing a function here as the argument? does it have to do with boundaries or unit testing?

Reply:
Domain layer should does not depend on service layer. Domain layer is made up of pure functional components. App layer is where we depend on external services (ie. not purely functional) and compose the domain layer primitives to use cases.

The more business logic we can extract into domain layer the better. Because they are pure functions we can unit test them which is simpler, faster and more isolated than the currently dominating integration test style we have for most of the functionality.

So it has to do with both boundaries and testing.

Boundaries because this is business logic that we would prefer to have in domain layer.
Testing because extracting into domain layer makes it a pure function.